src/app/courses/courses-search/courses-search.component.ts

```diff
@@ -1,18 +1,25 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
 
 @Component({
   selector: 'courses-search',
   templateUrl: './courses-search.component.html',
   styleUrls: ['./courses-search.component.scss'],
 })
 export class CoursesSearchComponent implements OnInit {
+  @Output() search = new EventEmitter<string>();
   public searchValue: string;
```

Comment:
You don't need an extra field here. You can get the search value from the event.

Reply:
Replaced `ngModel` with form to cover both (click and enter) submit cases